cpp/include/cudf/column/column_view.hpp

```diff
@@ -322,6 +322,18 @@ class column_view : public detail::column_view_base {
               size_type offset                         = 0,
               std::vector<column_view> const& children = {});
 
+  /**
+   * @brief Zero-copy cast between binary compatible types
+   *
+   * Can be used to cast timestamp columns to their underlying integer representation
+   *
+   * @throws cudf::logic_error if the specified type is not compatible
+   *
+   * @param type The `data_type` to cast to
+   * @return Column view with the specified type
+   */
+  column_view logical_cast(data_type type) const;
```

Comment:
Does this need to be a member function? I.e., does it need access to private members? Otherwise, prefer it to be a free function.

Reply:
It accesses _all_ the private members... 😅 I could make it a friend function if that would be preferable?
```cpp
column_view column_view::logical_cast(data_type type) const
{
  CUDF_EXPECTS(is_logically_castable(_type, type), "types are not logically castable");
  return column_view{type, _size, _data, _null_mask, _null_count, _offset, _children};
}
```

Reply:
Hm, if this _were_ a free function, would it make sense to move it to `unary.hpp` alongside `cudf::cast`?

Reply:
I went ahead and made this a friend function, but left it in `column_view.hpp` for now.